onboarding/main.go

```diff
@@ -18,80 +21,131 @@ import (
 )
 
 const (
-	onboardingTicketPublicKeySecretName = "onboarding-ticket-key" //Name of existing public key which is used ocs-operator
-	onboardingPrivateKeySecretName      = "onboarding-private-key"
-	serviceAccountName                  = "onboarding-secret-generator"
+	// Name of existing public key which is used ocs-operator
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+	onboardingValidationPrivateKeySecretName = "onboarding-private-key"
 )
 
 func main() {
 	clientset, err := newClient()
 	if err != nil {
-		klog.Error(err, "failed to create controller-runtime client")
-		return
+		klog.Errorf("failed to create clientset: %v", err)
+		os.Exit(1)
 	}
 
 	operatorNamespace, err := util.GetOperatorNamespace()
 	if err != nil {
-		klog.Error(err, "unable to get operator namespace")
+		klog.Errorf("unable to get operator namespace: %v", err)
+		os.Exit(1)
+	}
+
+	// Generate RSA key.
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		klog.Errorf("unable to generate private: %v", err)
+		os.Exit(1)
+	}
+
+	publicKey := &privateKey.PublicKey
+	// Export the keys to pem string
+	privatePem := convertRsaPrivateKeyAsPemStr(privateKey)
+	publicPem, err := convertRsaPublicKeyAsPemStr(publicKey)
+
+	if err != nil {
+		klog.Errorf("failed to convert public key to pem str: %v", err)
+		os.Exit(1)
+	}
+
+	sc := ocsv1.StorageClusterList{}
+	instance, err := clientset.RESTClient().Get().AbsPath(
```

Comment:
Initially we were only using kube native resources (secrets) and so was the comment that client-go is sufficient. 

It is evident now that runtime client abstraction is helpful and so you should be able to freely use it.

Reply:
I agree, but please do not do the change now,  we can move back to runtime client in a later PR

Reply:
I agree.